File: src/main/java/com/dafttech/terra/engine/renderer/SubtileRenderer.java
package com.dafttech.terra.engine.renderer;

import com.dafttech.terra.engine.AbstractScreen;
import com.dafttech.terra.game.world.entities.Entity;
import com.dafttech.terra.game.world.subtiles.Subtile;

public abstract class SubtileRenderer {
    public abstract void draw(AbstractScreen screen, Subtile render, Entity pointOfView, Object... rendererArguments);
}

File: terra/src/main/scala/com/dafttech/terra/engine/renderer/SubtileRenderer.scala
package com.dafttech.terra.engine.renderer

import com.dafttech.terra.engine.{AbstractScreen, TilePosition}
import com.dafttech.terra.game.world.entities.Entity
import com.dafttech.terra.game.world.subtiles.Subtile

abstract class SubtileRenderer {
  def draw(screen: AbstractScreen, render: Subtile, pointOfView: Entity, rendererArguments: AnyRef*)(implicit tp: TilePosition)
}